centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/Header/Actions/Delete.tsx

```diff
@@ -29,25 +32,32 @@ const DeleteAction = (): JSX.Element => {
   const getEndpoint = (): string =>
     notificationEndpoint({ id: notificationId });
 
-  const onSuccess = (): void => {
-    setPanelOpen(false);
-  };
-
   const {
     values: { name }
   } = useFormikContext<FormikValues>();
 
+  const { onClick, dialogOpen, isMutating, onCancel, onConfirm } = useDelete({
+    getEndpoint,
+    labelFailed: labelFailedToDeleteNotification,
+    labelSuccess: labelNotificationSuccessfullyDeleted,
+    onSuccess: () => setPanelOpen(false)
+  });
+
   return (
-    <DeleteButton
-      ariaLabel="delete a notification"
-      fetchMethod={Method.DELETE}
-      getEndpoint={getEndpoint}
-      iconClassName={classes.icon}
-      labelFailed={labelFailedToDeleteNotification}
-      labelSuccess={labelNotificationSuccessfullyDeleted}
-      notificationName={name}
-      onSuccess={onSuccess}
-    />
+    <Box>
+      <DeleteButton
+        ariaLabel={t(labelDeleteNotification) as string}
+        iconClassName={classes.icon}
+        onClick={onClick}
+      />
+      <ConfirmationDialog
+        dialogOpen={dialogOpen}
+        isMutating={isMutating}
+        notificationName={name}
+        onCancel={onCancel}
+        onConfirm={onConfirm}
+      />
```

Comment:
1. the `ConfirmationDialog` is still added multiple times (in edit panel + listing actions), could it be moved up to page level ?
2. the props on the `ConfirmationDialog` are not needed if it's using the `useDelete` hook itself to get this information (might be a good idea to be rename it to the specific use case → `DeleteConfirmationDialog`)

_the specifics (`notificationName` or notification object) can be passed with the `deleteNotification` action on the `onClick` handler of the `DeleteButton`_

Reply:
The issue I've encountered is that the useDelete hook requires parameters because the request and endpoint are not the same. For the listing header, the request for deleting multiple notifications is a POST request instead of a DELETE request. Additionally, we have different success/error messages, and for the action in the panel, we need to exit the panel upon success. These parameters must be specified in each part; otherwise, I would need to use three buttons and three files for requests.

Reply:
the hook could expose a `deleteSingleItem` and `deleteMultipleItems`, depending on this, the hook would be in another state, the dialog can show another variant (based on the hook state), and the `submit` action of the hook can have the appropriate POST / DELETE request